pkg/instancegroups/instancegroups.go

```diff
@@ -143,7 +203,73 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 		}
 	}
 
-	for _, u := range update {
+	count := 0
+	batchSize := 1
+	if featureflag.BatchRollingUpdate.Enabled() {
+		batchSize = rollingUpdateData.BatchSize
+	}
+
+	for count < len(update)/batchSize {
+		batchStart := count * batchSize
+		batchEnd := batchStart + batchSize
+		err = r.batchUpdate(rollingUpdateData, cluster, instanceGroupList, isBastion, sleepAfterTerminate, validationTimeout, update, batchStart, batchEnd)
+		if err != nil {
+			return err
+		}
+		count++
+	}
+
+	batchStart := count * batchSize
+	batchEnd := batchStart + len(update)%batchSize
+	err = r.batchUpdate(rollingUpdateData, cluster, instanceGroupList, isBastion, sleepAfterTerminate, validationTimeout, update, batchStart, batchEnd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *RollingUpdateInstanceGroup) batchUpdate(rollingUpdateData *RollingUpdateCluster, cluster *api.Cluster, instanceGroupList *api.InstanceGroupList, isBastion bool, sleepAfterTerminate time.Duration, validationTimeout time.Duration, update []*cloudinstances.CloudInstanceGroupMember, batchStart int, batchEnd int) (err error) {
+
+	if rollingUpdateData.Detach {
+
+		for i, u := range update[batchStart:batchEnd] {
+			instanceId := u.ID
+
+			nodeName := ""
+			if u.Node != nil {
+				nodeName = u.Node.Name
+			}
+			if err = r.DetachInstance(u); err != nil {
+				klog.Errorf("error deleting instance %q, node %q: %v", instanceId, nodeName, err)
+				return err
```

Comment:
...leaving all the earlier instances in the batch detached and unlikely to ever be terminated?